tests/test_scrape_github_trending.py

```diff
@@ -102,3 +102,88 @@ def test_fake_response_raises_on_error():
     response = FakeResponse(status_code=500)
     with pytest.raises(Exception):
         response.raise_for_status()
+
+
+def test_scrape_github_trending_skips_duplicates_precheck(monkeypatch):
```

Comment:
Can we add type hints to the new tests/helpers to keep tests aligned with PYTHON.md? Right now the new test functions and stubs are unannotated.